### PR TITLE
Restore url routing for directions

### LINKED ISSUE #574: Remove City Hall as default origin
Just leave blank.

### PR DESCRIPTION
**Pull request by KlaasH**
Re-enables reading and updating of URL parameters for interactive directions.

Also converts the UserPreferences component to just store values in a variable rather than in user storage, as a very rough first pass at issue #589.  Maybe this will be fine, maybe we still want to put some things in local storage, maybe things break, or maybe we want to go further and not use UserPreferences as an interface for sharing parameters between components.  Anyway, doing this was enough to get URL routing working.

I also moved the `home` view from the `cms` app to the `destinations` app because it seemed to fit in better there, and spruced up the shared view logic in `destinations/views.py` a bit.

And since I was in UserPreferences, I removed the City Hall default (resolves #574).

**To test**
- Following a link to directions should take you to the directions, e.g. http://localhost:8024/?origin=39.9616133%2C-75.1541914&originText=990%20Spring%20Garden%20St%2C%20Philadelphia%2C%20Pennsylvania%2C%2019123&mode=WALK%2CTRANSIT&destination=39.96581%2C-75.183494&destinationText=Fairmount%20Waterworks%20Interpretive%20Center
- Following the same link in the old format (with `map/directions` at the beginning) should redirect you to `/` and work, e.g. http://localhost:8024/map/directions?origin=39.9616133%2C-75.1541914&originText=990%20Spring%20Garden%20St%2C%20Philadelphia%2C%20Pennsylvania%2C%2019123&mode=WALK%2CTRANSIT&destination=39.96581%2C-75.183494&destinationText=Fairmount%20Waterworks%20Interpretive%20Center
- Since there's no more local storage, clearing the URL (i.e. going to http://localhost:8024/) should get you back to the home view with empty origin/destination and the default mode options.

### CODE REVIEW

**Comment by KlaasH** [comment]
Build failure was from an insufficiently escaped regex. I rebased the fix in.

File: python/cac_tripplanner/destinations/views.py [hunk 1]
```diff
@@ -1,54 +1,59 @@
...
+        context = {'tab': 'map'}
+    elif request.GET.get('origin') is not None:
+        # If there's no destination but there is an origin, go to Explore
+        context = {'tab': 'explore'}
+    else:
+        # Otherwise show the home view
+        # Show one random article
+        article = Article.objects.random()
+        # Show all destinations, in random order
+        destinations = Destination.objects.order_by('?').all()
```
Comment (KlaasH) [inline python/cac_tripplanner/destinations/views.py]: The note on `place_detail` reminded me: this resolves issue #585.

File: python/cac_tripplanner/destinations/views.py [hunk 2]
```diff
@@ -58,7 +63,15 @@ def directions(request):
     :param request: Request object
     :returns: A rendered response
     """
-    return base_otp_view(request, 'directions.html')
+    return base_view(request, 'directions.html')
+
+
+def place_detail(request, pk):
+    destination = get_object_or_404(Destination.objects.published(), pk=pk)
+    more_destinations = Destination.objects.published().order_by('?').exclude(pk=destination.pk)[:3]
```
Comment (KlaasH) [inline python/cac_tripplanner/destinations/views.py]: Not something I remember off the top of my head, but I just tried in shell_plus and learned that, while index lookups get grumpy when they're out of range, slices are forgiving.

**Comment by flibbertigibbet** [comment]
build again

**Pull request merged**